Frontend/js/raytracing.js

```diff
@@ -12,12 +12,48 @@ function addShape() {
 		//get coordinates(circle center, cube upper left corner
 		var x = parseInt(document.getElementById('shape_x').value);
 		var y = parseInt(document.getElementById('shape_y').value);
+		var z = parseInt(document.getElementById('shape_z').value);
+		
+		
+		//coordiates verification start (Farhad)
+		
+		if ( (x < 100) || ( isNaN(x) ) )

```

Comment:
Please refactor your code, you have the same if statements twice.

Reply:
Dear Sebastian, as a general advice, I would encourage you to read everybody's code very carefully before making any comments.

The two coordinate verification are one for light and one for shape and have nothing to do with each other.

furthermore, I would think it is inappropriate for anyone to comment on anything but functionality of other's code, because I would want to see a proper use of one of the industry standards like bootstrap in out front-end development but I respected chikans design and accepted it.

Reply:
some of the conflicts are because our dear friend chikans did the part that was assigned to me and I believe that is something that should be discussed in the group meeting.

Reply:
Dear Farhad, thank you for your advice. Unfortunately, I disagree with you about the difference of the verifications. The if-statements do exactly the same, independent from the type of object, light or shape. Maybe you can add a method to have the code only once?

Why do you think it is inappropriate to comment the source code of other team members? I think it is important, to discuss ideas and solutions. If you have some ideas, about the frontend and how to improve it, speak to @chiknas.

Reply:
Perhaps I was not too clear in my comment. 
The inappropriate part was the tone you used. 

If you believed a function would be a good Idea, you should have mentioned something like. "I believe it would be better if you used a general validation function".

As far as my opinion goes, I expressed it very clearly from the beginning if we are going to have drag and drop design, there is no need for using input values but we clearly didn't do that and I respected the group decision.